src/main/kotlin/com/demonwav/mcdev/platform/forge/reflection/reference/ReflectedFieldReference.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2018 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.forge.reflection.reference
+
+import com.demonwav.mcdev.facet.MinecraftFacet
+import com.demonwav.mcdev.platform.mcp.McpModuleType
+import com.demonwav.mcdev.util.constantStringValue
+import com.demonwav.mcdev.util.findModule
+import com.demonwav.mcdev.util.simpleQualifiedMemberReference
+import com.intellij.codeInsight.completion.JavaLookupElementBuilder
+import com.intellij.codeInsight.hint.HintManager
+import com.intellij.lang.jvm.JvmModifier
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.openapi.command.CommandProcessor
+import com.intellij.psi.JavaPsiFacade
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiClassObjectAccessExpression
+import com.intellij.psi.PsiClassType
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementResolveResult
+import com.intellij.psi.PsiExpressionList
+import com.intellij.psi.PsiLiteral
+import com.intellij.psi.PsiReference
+import com.intellij.psi.PsiReferenceBase
+import com.intellij.psi.PsiReferenceProvider
+import com.intellij.psi.ResolveResult
+import com.intellij.psi.codeStyle.CodeStyleManager
+import com.intellij.util.ProcessingContext
+
+object ReflectedFieldReference : PsiReferenceProvider() {
+    override fun getReferencesByElement(element: PsiElement, context: ProcessingContext): Array<PsiReference> {
+        // The pattern for this provider should only match method params, but better be safe
+        if (element.parent !is PsiExpressionList) {
+            return arrayOf()
+        }
+        return arrayOf(Reference(element as PsiLiteral))
+    }
+
+    class Reference(element: PsiLiteral) : PsiReferenceBase.Poly<PsiLiteral>(element) {
+        val fieldName
+            get() = element.constantStringValue ?: ""
+
+        override fun getVariants(): Array<Any> {
+            val typeClass = findReferencedClass() ?: return arrayOf()
+
+            return typeClass.allFields.filter { !it.hasModifier(JvmModifier.PUBLIC) }
```

Comment:
Sorry, don't have my IDE open right now. What does `allFields` return? If it's anything other than a `Stream` or `Sequence`, you need to add `.asSequence()` after `allFields`. Otherwise every operation creates a new collection.